http-client/src/main/java/io/airlift/http/client/RetryDriver.java

```diff
@@ -29,7 +31,7 @@
     private final double scaleFactor;
     private final Duration maxRetryTime;
     private final Function<Exception, Exception> exceptionMapper;
-    private final List<Class<? extends Exception>> exceptionWhiteList;
+    private final List<Predicate<Exception>> stopConditions;
```

Comment:
What about using this as a RetryDriver https://gist.github.com/findepi/b6b1eac05f4ed1e918341a78c7b28119 ?
Notable changes:
- lay the fields out in the order of they meaning (so stop/continue condition comes before delay timing)
- single easy to understand condition: `Predicate<Throwable> retryCondition` 
   - but `stopOn` does not accumulate as in Presto Hive connector RetryDriver
- `retryRunnable` is no longer optional
- null checks for everything that could be null